lib/flake-utils.nix

```diff
@@ -0,0 +1,63 @@
+{ nixpkgs, systems }:
+rec {
+  /* eachDefaultSystem using defaultSystems
+     Usage in a nix flake:
+       {
+         description = "Flake demo";
+
+         inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+         outputs = { self, nixpkgs, flake-utils }:
+           nixpkgs.lib.eachDefaultSystem (pkgs:
+             rec {
+               packages = {
+                 hello = pkgs.hello;
+               };
+               defaultPackage = packages.hello;
+             }
+           );
+       }
+
+     Type: eachDefaultSystem :: (attrset -> attrset) -> attrset
+
+     Example:
+       nixpkgs.lib.eachDefaultSystem (pkgs: {
+         defaultPackage = packages.hello;
+       })
+       => {
+         defaultPackage = {
+           aarch64-linux.hello = <<derivation /nix/store/...-hello-2.10.drv>>,
+           ...
+           x86_64-darwin.hello = <<derivation /nix/store/...-hello-2.10.drv>>,
+           x86_64-linux.hello = <<derivation /nix/store/...-hello-2.10.drv>>,
+         };
+       }
+  */
+  eachDefaultSystem = eachSystem systems;
+
+  /* Builds a map from <attr>=value to <attr>.<system>=value for each system.
+     See `eachDefaultSystem` for a full example
+
+     Type: eachSystem :: [string] -> (attrset -> attrset)
+
+     Example:
+       eachSystem ["x86_64-linux"] (pkgs: { hello = 42; })
+       => { hello = { x86_64-linux = 42; }; }
+  */
+  eachSystem = systems: f:
+    let
+      op = attrs: system:
+        let
+          pkgs = nixpkgs.legacyPackages.${system};
```

Comment:
This doesn't solve the general "problem" of getting things out of a flake. `nixpkgs` isn't supposed to be that different from other flakes. You could depend on two versions of nixpkgs simultaneously, etc.

What if we work with `self` instead and do this for all `self.inputs`?

Reply:
Please elaborate, what you mean.